pkg/codegen/docs/gen.go

```diff
@@ -2590,14 +2592,41 @@ func Initialize(tool string, pkg *schema.Package) {
 	defaultContext.initialize(tool, pkg)
 }
 
-// GeneratePackage generates docs for each resource given the Pulumi
-// schema. The returned map contains the filename with path as the key
-// and the contents as its value.
-func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error) {
-	return defaultContext.generatePackage(tool, pkg)
+// Context provides ambient context for docs generation.
+//
+// Context should be created with [NewContext]. The zero value of Context is not safe to use.
+type Context struct{ *docGenContext }
```

Comment:
I have probably missed the point here, but why are we not e.g. renaming `docGenContext` to make it public and just exposing that? Is it explicitly the "we don't want you messing around with private unsafe values" part?

Reply:
2 benefits:
1. This makes checking if a `Context` was initialized trivial.
2. The caller of `NewContext` doesn't need to wander what the pointer means in `NewContext(...) *Context`.